{
    "1.0": {
        "info": {
            "author": "\u7af9\u6c38\u5eb7",
            "author_email": "gqylpy@outlook.com",
            "bugtrack_url": null,
            "classifiers": [
                "Environment :: Web Environment",
                "Intended Audience :: Developers",
                "License :: OSI Approved :: Apache Software License",
                "Operating System :: OS Independent",
                "Programming Language :: Python :: 3.10",
                "Programming Language :: Python :: 3.6",
                "Programming Language :: Python :: 3.7",
                "Programming Language :: Python :: 3.8",
                "Programming Language :: Python :: 3.9",
                "Topic :: Internet",
                "Topic :: Software Development :: Libraries :: Python Modules",
                "Topic :: Text Processing :: Indexing",
                "Topic :: Utilities"
            ],
            "description_content_type": "text/markdown",
            "docs_url": null,
            "download_url": "",
            "downloads": {
                "last_day": -1,
                "last_month": -1,
                "last_week": -1
            },
            "home_page": "http://gqylpy.com",
            "keywords": "",
            "license": "Apache 2.0",
            "maintainer": "",
            "maintainer_email": "",
            "name": "gqylpy-exception",
            "package_url": "https://pypi.org/project/gqylpy-exception/",
            "platform": null,
            "project_url": "https://pypi.org/project/gqylpy-exception/",
            "project_urls": {
                "Homepage": "http://gqylpy.com",
                "Source": "https://github.com/gqylpy/gqylpy-exception"
            },
            "release_url": "https://pypi.org/project/gqylpy-exception/1.0/",
            "requires_dist": null,
            "requires_python": ">=3.6",
            "summary": "",
            "version": "1.0",
            "yanked": false,
            "yanked_reason": null
        },
        "last_serial": 15750466,
        "urls": [
            {
                "comment_text": "",
                "digests": {
                    "md5": "48fd82ef19c9d78b44eac0986f8152ee",
                    "sha256": "0dc535ffaec8d441200d4a3560a95fd11872103e537fa39827849e207a72badd"
                },
                "downloads": -1,
                "filename": "gqylpy_exception-1.0-py3-none-any.whl",
                "has_sig": false,
                "md5_digest": "48fd82ef19c9d78b44eac0986f8152ee",
                "packagetype": "bdist_wheel",
                "python_version": "py3",
                "requires_python": ">=3.6",
                "size": 7867,
                "upload_time": "2022-05-31T05:53:59",
                "upload_time_iso_8601": "2022-05-31T05:53:59.798324Z",
                "url": "https://files.pythonhosted.org/packages/3c/f8/e7e0afa56f653f441972d046af8bc72c88c3437f6f0a89beec6c6a0cfae1/gqylpy_exception-1.0-py3-none-any.whl",
                "yanked": false,
                "yanked_reason": null
            }
        ],
        "vulnerabilities": []
    },
    "1.1": {
        "info": {
            "author": "\u7af9\u6c38\u5eb7",
            "author_email": "gqylpy@outlook.com",
            "bugtrack_url": null,
            "classifiers": [
                "Environment :: Web Environment",
                "Intended Audience :: Developers",
                "License :: OSI Approved :: Apache Software License",
                "Operating System :: OS Independent",
                "Programming Language :: Python :: 3.10",
                "Programming Language :: Python :: 3.6",
                "Programming Language :: Python :: 3.7",
                "Programming Language :: Python :: 3.8",
                "Programming Language :: Python :: 3.9",
                "Topic :: Internet",
                "Topic :: Software Development :: Libraries :: Python Modules",
                "Topic :: Text Processing :: Indexing",
                "Topic :: Utilities"
            ],
            "description_content_type": "text/markdown",
            "docs_url": null,
            "download_url": "",
            "downloads": {
                "last_day": -1,
                "last_month": -1,
                "last_week": -1
            },
            "home_page": "http://gqylpy.com",
            "keywords": "",
            "license": "Apache 2.0",
            "maintainer": "",
            "maintainer_email": "",
            "name": "gqylpy-exception",
            "package_url": "https://pypi.org/project/gqylpy-exception/",
            "platform": null,
            "project_url": "https://pypi.org/project/gqylpy-exception/",
            "project_urls": {
                "Homepage": "http://gqylpy.com",
                "Source": "https://github.com/gqylpy/gqylpy-exception"
            },
            "release_url": "https://pypi.org/project/gqylpy-exception/1.1/",
            "requires_dist": null,
            "requires_python": ">=3.6",
            "summary": "",
            "version": "1.1",
            "yanked": false,
            "yanked_reason": null
        },
        "last_serial": 15750466,
        "urls": [
            {
                "comment_text": "",
                "digests": {
                    "md5": "b505a8dfcf868aee91b2f367c53e983d",
                    "sha256": "8322f5b96bf81549f6f022ae54f69e663381ca2e96997c9690c720c9cc49e6b1"
                },
                "downloads": -1,
                "filename": "gqylpy_exception-1.1-py3-none-any.whl",
                "has_sig": false,
                "md5_digest": "b505a8dfcf868aee91b2f367c53e983d",
                "packagetype": "bdist_wheel",
                "python_version": "py3",
                "requires_python": ">=3.6",
                "size": 12096,
                "upload_time": "2022-06-22T11:34:28",
                "upload_time_iso_8601": "2022-06-22T11:34:28.875349Z",
                "url": "https://files.pythonhosted.org/packages/1d/cf/30fb23b9e3db3b03a9c390d5ba14ada0b9e63ea337d9c31eedeafe6232a3/gqylpy_exception-1.1-py3-none-any.whl",
                "yanked": false,
                "yanked_reason": null
            },
            {
                "comment_text": "",
                "digests": {
                    "md5": "a8bae7c887bf579daede550b4f5c40f5",
                    "sha256": "a911a86f69ab121a250998722460efb1568fc6e33374a4a44effe8924ca4be72"
                },
                "downloads": -1,
                "filename": "gqylpy_exception-1.1.tar.gz",
                "has_sig": false,
                "md5_digest": "a8bae7c887bf579daede550b4f5c40f5",
                "packagetype": "sdist",
                "python_version": "source",
                "requires_python": ">=3.6",
                "size": 12730,
                "upload_time": "2022-06-22T11:34:30",
                "upload_time_iso_8601": "2022-06-22T11:34:30.530286Z",
                "url": "https://files.pythonhosted.org/packages/31/81/652f8d238599cd7e1fa1abad97cce11238e3b135f16b47a8d2500ca18ae0/gqylpy_exception-1.1.tar.gz",
                "yanked": false,
                "yanked_reason": null
            }
        ],
        "vulnerabilities": []
    },
    "1.1.1": {
        "info": {
            "author": "\u7af9\u6c38\u5eb7",
            "author_email": "gqylpy@outlook.com",
            "bugtrack_url": null,
            "classifiers": [
                "Environment :: Web Environment",
                "Intended Audience :: Developers",
                "License :: OSI Approved :: Apache Software License",
                "Operating System :: OS Independent",
                "Programming Language :: Python :: 3.10",
                "Programming Language :: Python :: 3.6",
                "Programming Language :: Python :: 3.7",
                "Programming Language :: Python :: 3.8",
                "Programming Language :: Python :: 3.9",
                "Topic :: Internet",
                "Topic :: Software Development :: Libraries :: Python Modules",
                "Topic :: Text Processing :: Indexing",
                "Topic :: Utilities"
            ],
            "description_content_type": "text/markdown",
            "docs_url": null,
            "download_url": "",
            "downloads": {
                "last_day": -1,
                "last_month": -1,
                "last_week": -1
            },
            "home_page": "http://gqylpy.com",
            "keywords": "",
            "license": "Apache 2.0",
            "maintainer": "",
            "maintainer_email": "",
            "name": "gqylpy-exception",
            "package_url": "https://pypi.org/project/gqylpy-exception/",
            "platform": null,
            "project_url": "https://pypi.org/project/gqylpy-exception/",
            "project_urls": {
                "Homepage": "http://gqylpy.com",
                "Source": "https://github.com/gqylpy/gqylpy-exception"
            },
            "release_url": "https://pypi.org/project/gqylpy-exception/1.1.1/",
            "requires_dist": null,
            "requires_python": ">=3.6",
            "summary": "",
            "version": "1.1.1",
            "yanked": false,
            "yanked_reason": null
        },
        "last_serial": 15750466,
        "urls": [
            {
                "comment_text": "",
                "digests": {
                    "md5": "c82f1f8098020438911a012fd0cbafa0",
                    "sha256": "ae926d09fcd6e791ae8f8596da935f5f9b876c497fb84743d76c003a3229ae20"
                },
                "downloads": -1,
                "filename": "gqylpy_exception-1.1.1-py3-none-any.whl",
                "has_sig": false,
                "md5_digest": "c82f1f8098020438911a012fd0cbafa0",
                "packagetype": "bdist_wheel",
                "python_version": "py3",
                "requires_python": ">=3.6",
                "size": 12145,
                "upload_time": "2022-07-27T09:36:18",
                "upload_time_iso_8601": "2022-07-27T09:36:18.020165Z",
                "url": "https://files.pythonhosted.org/packages/5a/ec/97cd9d10ff2cd95294adf386a366b27eab21a87fddc820d97b3c167c90d5/gqylpy_exception-1.1.1-py3-none-any.whl",
                "yanked": false,
                "yanked_reason": null
            },
            {
                "comment_text": "",
                "digests": {
                    "md5": "0da46d52037f08e649c8dc6d58bcf937",
                    "sha256": "a4eeeb9cb8a82855182666ec5847514805fc4e6bbd18c98e3dfe3211c48cacce"
                },
                "downloads": -1,
                "filename": "gqylpy_exception-1.1.1.tar.gz",
                "has_sig": false,
                "md5_digest": "0da46d52037f08e649c8dc6d58bcf937",
                "packagetype": "sdist",
                "python_version": "source",
                "requires_python": ">=3.6",
                "size": 12758,
                "upload_time": "2022-07-27T09:36:19",
                "upload_time_iso_8601": "2022-07-27T09:36:19.801397Z",
                "url": "https://files.pythonhosted.org/packages/a6/01/9ebedf17dfab887fd5c9b53da7429ef75af6e2ff2bdadd6a2616732c24dc/gqylpy_exception-1.1.1.tar.gz",
                "yanked": false,
                "yanked_reason": null
            }
        ],
        "vulnerabilities": []
    },
    "1.1.2": {
        "info": {
            "author": "\u7af9\u6c38\u5eb7",
            "author_email": "gqylpy@outlook.com",
            "bugtrack_url": null,
            "classifiers": [
                "Environment :: Web Environment",
                "Intended Audience :: Developers",
                "License :: OSI Approved :: Apache Software License",
                "Operating System :: OS Independent",
                "Programming Language :: Python :: 3.10",
                "Programming Language :: Python :: 3.6",
                "Programming Language :: Python :: 3.7",
                "Programming Language :: Python :: 3.8",
                "Programming Language :: Python :: 3.9",
                "Topic :: Internet",
                "Topic :: Software Development :: Libraries :: Python Modules",
                "Topic :: Text Processing :: Indexing",
                "Topic :: Utilities"
            ],
            "description_content_type": "text/markdown",
            "docs_url": null,
            "download_url": "",
            "downloads": {
                "last_day": -1,
                "last_month": -1,
                "last_week": -1
            },
            "home_page": "http://gqylpy.com",
            "keywords": "",
            "license": "Apache 2.0",
            "maintainer": "",
            "maintainer_email": "",
            "name": "gqylpy-exception",
            "package_url": "https://pypi.org/project/gqylpy-exception/",
            "platform": null,
            "project_url": "https://pypi.org/project/gqylpy-exception/",
            "project_urls": {
                "Homepage": "http://gqylpy.com",
                "Source": "https://github.com/gqylpy/gqylpy-exception"
            },
            "release_url": "https://pypi.org/project/gqylpy-exception/1.1.2/",
            "requires_dist": null,
            "requires_python": ">=3.6",
            "summary": "",
            "version": "1.1.2",
            "yanked": false,
            "yanked_reason": null
        },
        "last_serial": 15750466,
        "urls": [
            {
                "comment_text": "",
                "digests": {
                    "md5": "c4b7124b1330a3f85ebe79abd4dcbcd8",
                    "sha256": "6246ba5eba194b8348b3aa41a3483f0281bcbeb6b7be6a6482b9e5add6e85cc2"
                },
                "downloads": -1,
                "filename": "gqylpy_exception-1.1.2-py3-none-any.whl",
                "has_sig": false,
                "md5_digest": "c4b7124b1330a3f85ebe79abd4dcbcd8",
                "packagetype": "bdist_wheel",
                "python_version": "py3",
                "requires_python": ">=3.6",
                "size": 12142,
                "upload_time": "2022-08-03T10:15:01",
                "upload_time_iso_8601": "2022-08-03T10:15:01.974251Z",
                "url": "https://files.pythonhosted.org/packages/73/ef/820ca0e3b4811f03ef237d46f089450f827a9a71f91bf3d631c7cecc0055/gqylpy_exception-1.1.2-py3-none-any.whl",
                "yanked": false,
                "yanked_reason": null
            },
            {
                "comment_text": "",
                "digests": {
                    "md5": "7a5f44bfbba03d294ffbaf786d4432d0",
                    "sha256": "98983024a97b0c656afa629ea0a7164c393b237a0da81d8a69ec1f07711f2ee2"
                },
                "downloads": -1,
                "filename": "gqylpy_exception-1.1.2.tar.gz",
                "has_sig": false,
                "md5_digest": "7a5f44bfbba03d294ffbaf786d4432d0",
                "packagetype": "sdist",
                "python_version": "source",
                "requires_python": ">=3.6",
                "size": 12769,
                "upload_time": "2022-08-03T10:15:03",
                "upload_time_iso_8601": "2022-08-03T10:15:03.148903Z",
                "url": "https://files.pythonhosted.org/packages/af/25/f2a58e01602c70d34e47fc16d3f673c4f96a792d6aaa7f81d9e856244f1c/gqylpy_exception-1.1.2.tar.gz",
                "yanked": false,
                "yanked_reason": null
            }
        ],
        "vulnerabilities": []
    },
    "1.1.3": {
        "info": {
            "author": "\u7af9\u6c38\u5eb7",
            "author_email": "gqylpy@outlook.com",
            "bugtrack_url": null,
            "classifiers": [
                "Environment :: Web Environment",
                "Intended Audience :: Developers",
                "License :: OSI Approved :: Apache Software License",
                "Operating System :: OS Independent",
                "Programming Language :: Python :: 3.10",
                "Programming Language :: Python :: 3.6",
                "Programming Language :: Python :: 3.7",
                "Programming Language :: Python :: 3.8",
                "Programming Language :: Python :: 3.9",
                "Topic :: Internet",
                "Topic :: Software Development :: Libraries :: Python Modules",
                "Topic :: Text Processing :: Indexing",
                "Topic :: Utilities"
            ],
            "description_content_type": "text/markdown",
            "docs_url": null,
            "download_url": "",
            "downloads": {
                "last_day": -1,
                "last_month": -1,
                "last_week": -1
            },
            "home_page": "http://gqylpy.com",
            "keywords": "",
            "license": "Apache 2.0",
            "maintainer": "",
            "maintainer_email": "",
            "name": "gqylpy-exception",
            "package_url": "https://pypi.org/project/gqylpy-exception/",
            "platform": null,
            "project_url": "https://pypi.org/project/gqylpy-exception/",
            "project_urls": {
                "Homepage": "http://gqylpy.com",
                "Source": "https://github.com/gqylpy/gqylpy-exception"
            },
            "release_url": "https://pypi.org/project/gqylpy-exception/1.1.3/",
            "requires_dist": null,
            "requires_python": ">=3.6",
            "summary": "",
            "version": "1.1.3",
            "yanked": false,
            "yanked_reason": null
        },
        "last_serial": 15750466,
        "urls": [
            {
                "comment_text": "",
                "digests": {
                    "md5": "5a21c3cde4cfd3d8d35c24164aa2ccc2",
                    "sha256": "a569c01042a2f818b279bc65ce1a8f1b9f97c5d81755012820958b8826521c60"
                },
                "downloads": -1,
                "filename": "gqylpy_exception-1.1.3-py3-none-any.whl",
                "has_sig": false,
                "md5_digest": "5a21c3cde4cfd3d8d35c24164aa2ccc2",
                "packagetype": "bdist_wheel",
                "python_version": "py3",
                "requires_python": ">=3.6",
                "size": 12191,
                "upload_time": "2022-10-07T10:38:03",
                "upload_time_iso_8601": "2022-10-07T10:38:03.892927Z",
                "url": "https://files.pythonhosted.org/packages/87/9f/f96cb2597a3163ec1fca0576880cea7c8ceea007547832de85b751f2a885/gqylpy_exception-1.1.3-py3-none-any.whl",
                "yanked": false,
                "yanked_reason": null
            },
            {
                "comment_text": "",
                "digests": {
                    "md5": "1e568d72a672f11c46eba2a5e2e7f901",
                    "sha256": "d5e4929db3295ee1bce0e9ef4119efe663c90f748371f6401ec8d72c1b44b32e"
                },
                "downloads": -1,
                "filename": "gqylpy_exception-1.1.3.tar.gz",
                "has_sig": false,
                "md5_digest": "1e568d72a672f11c46eba2a5e2e7f901",
                "packagetype": "sdist",
                "python_version": "source",
                "requires_python": ">=3.6",
                "size": 12798,
                "upload_time": "2022-10-07T10:38:05",
                "upload_time_iso_8601": "2022-10-07T10:38:05.406340Z",
                "url": "https://files.pythonhosted.org/packages/9a/97/3c552241022aa606bb8bd3b3a009dd9b1d17ec685af8b61420872dc85680/gqylpy_exception-1.1.3.tar.gz",
                "yanked": false,
                "yanked_reason": null
            }
        ],
        "vulnerabilities": []
    },
    "1.2": {
        "info": {
            "author": "\u7af9\u6c38\u5eb7",
            "author_email": "gqylpy@outlook.com",
            "bugtrack_url": null,
            "classifiers": [
                "Environment :: Web Environment",
                "Intended Audience :: Developers",
                "License :: OSI Approved :: Apache Software License",
                "Operating System :: OS Independent",
                "Programming Language :: Python :: 3.10",
                "Programming Language :: Python :: 3.6",
                "Programming Language :: Python :: 3.7",
                "Programming Language :: Python :: 3.8",
                "Programming Language :: Python :: 3.9",
                "Topic :: Internet",
                "Topic :: Software Development :: Libraries :: Python Modules",
                "Topic :: Text Processing :: Indexing",
                "Topic :: Utilities"
            ],
            "description_content_type": "text/markdown",
            "docs_url": null,
            "download_url": "",
            "downloads": {
                "last_day": -1,
                "last_month": -1,
                "last_week": -1
            },
            "home_page": "http://gqylpy.com",
            "keywords": "",
            "license": "Apache 2.0",
            "maintainer": "",
            "maintainer_email": "",
            "name": "gqylpy-exception",
            "package_url": "https://pypi.org/project/gqylpy-exception/",
            "platform": null,
            "project_url": "https://pypi.org/project/gqylpy-exception/",
            "project_urls": {
                "Homepage": "http://gqylpy.com",
                "Source": "https://github.com/gqylpy/gqylpy-exception"
            },
            "release_url": "https://pypi.org/project/gqylpy-exception/1.2/",
            "requires_dist": null,
            "requires_python": ">=3.6",
            "summary": "\u5728\u6267\u884c raise \u8bed\u53e5\u7684\u540c\u65f6\u521b\u5efa\u5f02\u5e38\u7c7b\uff0c\u65e0\u9700\u4e8b\u5148\u5b9a\u4e49\u5f02\u5e38\u7c7b\uff0c\u65b9\u4fbf\u5feb\u6377\u3002",
            "version": "1.2",
            "yanked": false,
            "yanked_reason": null
        },
        "last_serial": 15750466,
        "urls": [
            {
                "comment_text": "",
                "digests": {
                    "md5": "1f86d15fc7ebfa344c57e4e25e4d6a84",
                    "sha256": "da4a4d6cb8d34e0a6d5984b23f6c5a40d9e278e3c0c03dbb27face45ace047ae"
                },
                "downloads": -1,
                "filename": "gqylpy_exception-1.2-py3-none-any.whl",
                "has_sig": false,
                "md5_digest": "1f86d15fc7ebfa344c57e4e25e4d6a84",
                "packagetype": "bdist_wheel",
                "python_version": "py3",
                "requires_python": ">=3.6",
                "size": 12460,
                "upload_time": "2022-10-16T10:56:29",
                "upload_time_iso_8601": "2022-10-16T10:56:29.944642Z",
                "url": "https://files.pythonhosted.org/packages/75/2f/b37b100f5c2d6a347a91912d2f5c6aca1cc393687cef667b7694cdbeb105/gqylpy_exception-1.2-py3-none-any.whl",
                "yanked": false,
                "yanked_reason": null
            },
            {
                "comment_text": "",
                "digests": {
                    "md5": "e1f007739e7821882a02dd0da9ea481b",
                    "sha256": "7aae32670e330d18836a79825de83d81beb04ce972327850e0f3a668e0020d35"
                },
                "downloads": -1,
                "filename": "gqylpy_exception-1.2.tar.gz",
                "has_sig": false,
                "md5_digest": "e1f007739e7821882a02dd0da9ea481b",
                "packagetype": "sdist",
                "python_version": "source",
                "requires_python": ">=3.6",
                "size": 13091,
                "upload_time": "2022-10-16T10:56:31",
                "upload_time_iso_8601": "2022-10-16T10:56:31.609202Z",
                "url": "https://files.pythonhosted.org/packages/f6/9c/990c72432450eeb2943fe5772d9278043646b82de65054f2a1b770f472ca/gqylpy_exception-1.2.tar.gz",
                "yanked": false,
                "yanked_reason": null
            }
        ],
        "vulnerabilities": []
    },
    "1.2.1": {
        "info": {
            "author": "\u7af9\u6c38\u5eb7",
            "author_email": "gqylpy@outlook.com",
            "bugtrack_url": null,
            "classifiers": [
                "Environment :: Web Environment",
                "Intended Audience :: Developers",
                "License :: OSI Approved :: Apache Software License",
                "Operating System :: OS Independent",
                "Programming Language :: Python :: 3.10",
                "Programming Language :: Python :: 3.6",
                "Programming Language :: Python :: 3.7",
                "Programming Language :: Python :: 3.8",
                "Programming Language :: Python :: 3.9",
                "Topic :: Internet",
                "Topic :: Software Development :: Libraries :: Python Modules",
                "Topic :: Text Processing :: Indexing",
                "Topic :: Utilities"
            ],
            "description_content_type": "text/markdown",
            "docs_url": null,
            "download_url": "",
            "downloads": {
                "last_day": -1,
                "last_month": -1,
                "last_week": -1
            },
            "home_page": "http://gqylpy.com",
            "keywords": "",
            "license": "Apache 2.0",
            "maintainer": "",
            "maintainer_email": "",
            "name": "gqylpy-exception",
            "package_url": "https://pypi.org/project/gqylpy-exception/",
            "platform": null,
            "project_url": "https://pypi.org/project/gqylpy-exception/",
            "project_urls": {
                "Homepage": "http://gqylpy.com",
                "Source": "https://github.com/gqylpy/gqylpy-exception"
            },
            "release_url": "https://pypi.org/project/gqylpy-exception/1.2.1/",
            "requires_dist": null,
            "requires_python": ">=3.6",
            "summary": "\u5728\u6267\u884c raise \u8bed\u53e5\u7684\u540c\u65f6\u521b\u5efa\u5f02\u5e38\u7c7b\uff0c\u65e0\u9700\u4e8b\u5148\u5b9a\u4e49\u5f02\u5e38\u7c7b\uff0c\u65b9\u4fbf\u5feb\u6377\u3002",
            "version": "1.2.1",
            "yanked": false,
            "yanked_reason": null
        },
        "last_serial": 15750466,
        "urls": [
            {
                "comment_text": "",
                "digests": {
                    "md5": "9dabde45e7b29c6915b4f2fa044e4069",
                    "sha256": "ed1827361ccc2865927047a878029a8660040a3a57be317c3391783e64a94d88"
                },
                "downloads": -1,
                "filename": "gqylpy_exception-1.2.1-py3-none-any.whl",
                "has_sig": false,
                "md5_digest": "9dabde45e7b29c6915b4f2fa044e4069",
                "packagetype": "bdist_wheel",
                "python_version": "py3",
                "requires_python": ">=3.6",
                "size": 12507,
                "upload_time": "2022-10-23T10:09:50",
                "upload_time_iso_8601": "2022-10-23T10:09:50.533748Z",
                "url": "https://files.pythonhosted.org/packages/36/7e/718b9ba9fa76b7c141ffd47da67b972eb4ccc1507f7c6a039ba914cbd897/gqylpy_exception-1.2.1-py3-none-any.whl",
                "yanked": false,
                "yanked_reason": null
            },
            {
                "comment_text": "",
                "digests": {
                    "md5": "d61eca584dfa9e1dad0c1b85c3b74949",
                    "sha256": "83e40fba39c34b78687aa3ef37b4cc19f4384fa3883fca6e0517aff460cd9034"
                },
                "downloads": -1,
                "filename": "gqylpy_exception-1.2.1.tar.gz",
                "has_sig": false,
                "md5_digest": "d61eca584dfa9e1dad0c1b85c3b74949",
                "packagetype": "sdist",
                "python_version": "source",
                "requires_python": ">=3.6",
                "size": 13134,
                "upload_time": "2022-10-23T10:09:52",
                "upload_time_iso_8601": "2022-10-23T10:09:52.222299Z",
                "url": "https://files.pythonhosted.org/packages/e8/3b/98cb35376202fb251adad6b431d8e532e9af9a1f590de67a642f0eb87cba/gqylpy_exception-1.2.1.tar.gz",
                "yanked": false,
                "yanked_reason": null
            }
        ],
        "vulnerabilities": []
    },
    "1.3": {
        "info": {
            "author": "\u7af9\u6c38\u5eb7",
            "author_email": "<gqylpy@outlook.com>",
            "bugtrack_url": null,
            "classifiers": [
                "Environment :: Web Environment",
                "Intended Audience :: Developers",
                "License :: OSI Approved :: Apache Software License",
                "Operating System :: OS Independent",
                "Programming Language :: Python :: 3.10",
                "Programming Language :: Python :: 3.11",
                "Programming Language :: Python :: 3.6",
                "Programming Language :: Python :: 3.7",
                "Programming Language :: Python :: 3.8",
                "Programming Language :: Python :: 3.9",
                "Topic :: Internet",
                "Topic :: Software Development :: Libraries :: Python Modules",
                "Topic :: Text Processing :: Indexing",
                "Topic :: Utilities"
            ],
            "description": "[<img alt=\"LOGO\" src=\"http://www.gqylpy.com/static/img/favicon.ico\" height=\"21\" width=\"21\"/>](http://www.gqylpy.com)\n[![Release](https://img.shields.io/github/release/gqylpy/gqylpy-exception.svg?style=flat-square\")](https://github.com/gqylpy/gqylpy-exception/releases/latest)\n[![Python Versions](https://img.shields.io/pypi/pyversions/gqylpy_exception)](https://pypi.org/project/gqylpy_exception)\n[![License](https://img.shields.io/pypi/l/gqylpy_exception)](https://github.com/gqylpy/gqylpy-exception/blob/master/LICENSE)\n[![Downloads](https://pepy.tech/badge/gqylpy_exception/month)](https://pepy.tech/project/gqylpy_exception)\n\n# gqylpy-exception\n\n\n> \u5728\u6267\u884c `raise` \u8bed\u53e5\u7684\u540c\u65f6\u521b\u5efa\u5f02\u5e38\u7c7b\uff0c\u65e0\u9700\u4e8b\u5148\u5b9a\u4e49\u5f02\u5e38\u7c7b\uff0c\u65b9\u4fbf\u5feb\u6377\u3002\u4f8b\u5982\uff0c\u4f60\u60f3\u629b\u51fa\u4e00\u4e2a\u540d\u4e3a `NotUnderstandError` \u7684\u5f02\u5e38\uff0c\n> \u5bfc\u5165 `import gqylpy_exception as ge` \u540e\u76f4\u63a5\u6267\u884c `raise ge.NotUnderstandError` \u5373\u53ef\u3002\n> \n> `gqylpy-exception` \u8fd8\u63d0\u4f9b\u4e86\u4e24\u4e2a\u5904\u7406\u5f02\u5e38\u7684\u88c5\u9970\u5668\uff1a\n> - `TryExcept`: \u622a\u83b7\u88ab\u88c5\u9970\u7684\u51fd\u6570\u4e2d\u5f15\u53d1\u7684\u5f02\u5e38\uff0c\u5e76\u5c06\u5f02\u5e38\u4fe1\u606f\u8f93\u51fa\u5230\u7ec8\u7aef\uff0c\u4e0d\u662f\u629b\u51fa\u3002\n> - `Retry`: \u540c\u4e0a\uff0c\u5e76\u4f1a\u5c1d\u8bd5\u91cd\u65b0\u6267\u884c\uff0c\u901a\u8fc7\u53c2\u6570\u63a7\u5236\u6b21\u6570\uff0c\u5728\u8fbe\u5230\u6700\u5927\u6b21\u6570\u540e\u629b\u51fa\u5f02\u5e38\u3002\n\n<kbd>pip3 install gqylpy_exception</kbd>\n\n\n### \u4f7f\u7528 `gqylpy_exception` \u521b\u5efa\u5f02\u5e38\u7c7b\n```python\nimport gqylpy_exception as ge\n\nraise ge.AnError(...)\n```\n`gqylpy_exception` \u53ef\u4ee5\u521b\u5efa\u4efb\u610f\u540d\u79f0\u7684\u5f02\u5e38\u7c7b\u3002`AnError` \u4e0d\u662f `gqylpy_exception` \u4e2d\u5185\u7f6e\u7684\uff0c\u5b83\u662f\u5728\u4f60\u7684\u4ee3\u7801\u6267\u884c\u5230 `ge.` \n\u65f6\u521b\u5efa\u7684\uff0c\u9b54\u5316\u65b9\u6cd5 `__getattr__` \u7684\u7279\u6027\u3002\u4f60\u8fd8\u53ef\u4ee5\u901a\u8fc7\u9b54\u6cd5\u65b9\u6cd5 `__getitem__` \u83b7\u5f97\u5b83\uff1a\n```python\ne: ge.GqylpyError = ge['AnError'](...)\n```\n\u662f\u7684\uff0c\u4f7f\u7528 `gqylpy_exception` \u521b\u5efa\u7684\u5f02\u5e38\u7c7b\u90fd\u7ee7\u627f `GqylpyError`\uff0c`GqylpyError` \u7ee7\u627f\u5185\u7f6e\u7684 `Exception`\u3002\n\n\u8fd8\u6709\u4e00\u79cd\u7528\u6cd5\uff0c\u5bfc\u5165\u5373\u521b\u5efa\uff1a\n```python\nfrom gqylpy_exception import AnError\n\nraise AnError(...)\n```\n\u53e6\u5916\uff0c`gqylpy_exception` \u4e0d\u4f1a\u91cd\u590d\u521b\u5efa\u5f02\u5e38\u7c7b\uff0c\u521b\u5efa\u8fc7\u7684\u5f02\u5e38\u7c7b\u5c06\u5b58\u5165 `ge.__history__` \u5b57\u5178\uff0c\u5f53\u4f60\u518d\u6b21\u521b\u5efa\u65f6\u4ece\u8fd9\u4e2a\u5b57\u5178\u4e2d\u53d6\u503c\u3002\n\n\n### \u4f7f\u7528\u88c5\u9970\u5668 `TryExcept` \u5904\u7406\u51fd\u6570\u4e2d\u5f15\u53d1\u7684\u5f02\u5e38\n```python\nfrom gqylpy_exception import TryExcept\n\n@TryExcept(ValueError)\ndef func():\n    int('a')\n```\n\u9ed8\u8ba4\u7684\u5904\u7406\u6d41\u7a0b\u662f\u5c06\u5f02\u5e38\u7b80\u8981\u4fe1\u606f\u8f93\u51fa\u5230\u7ec8\u7aef\u3002\u5f53\u7136\uff0c\u4e5f\u53ef\u4ee5\u8f93\u51fa\u5230\u6587\u4ef6\u6216\u505a\u5176\u5b83\u5904\u7406\uff0c\u901a\u8fc7\u53c2\u6570\u63a7\u5236\uff1a\n```python\ndef TryExcept(\n        etype:          Union[type, Tuple[type, ...]],\n        /, *,\n        ignore:         bool                                       = False,\n        output_raw_exc: bool                                       = False,\n        logger:         logging.Logger                             = None,\n        ereturn:        Any                                        = None,\n        ecallback:      Callable[[Exception, Callable, ...], None] = None,\n        eexit:          bool                                       = False\n):\n    ...\n```\n__\u53c2\u6570 `etype`__<br>\n\u8981\u5904\u7406\u54ea\u79cd\u5f02\u5e38\uff0c\u4f7f\u7528\u5143\u7956\u4f20\u5165\u591a\u4e2a\u3002\n\n__\u53c2\u6570 `ingore`__<br>\n\u8bbe\u4e3a `True` \u5c06\u9759\u9ed8\u5904\u7406\u5f02\u5e38\uff0c\u6ca1\u6709\u4efb\u4f55\u8f93\u51fa\u3002\n\n__\u53c2\u6570 `output_raw_exc`__<br>\n\u8bbe\u4e3a `True` \u5c06\u8f93\u51fa\u5b8c\u6574\u7684\u5f02\u5e38\u4fe1\u606f\uff0c\u6ce8\u610f\u5176\u4f18\u5148\u7ea7\u4f4e\u4e8e `ignore`\u3002 \n\n__\u53c2\u6570 `logger`__<br>\n\u63a5\u6536\u4e00\u4e2a\u65e5\u5fd7\u8bb0\u5f55\u5668\u5bf9\u8c61\uff0c`TryExcept` \u5e0c\u671b\u4f7f\u7528\u65e5\u5fd7\u8bb0\u5f55\u5668\u8f93\u51fa\u5f02\u5e38\u4fe1\u606f\uff0c\u5b83\u8c03\u7528\u65e5\u5fd7\u8bb0\u5f55\u5668\u7684 `error` \u65b9\u6cd5\u3002<br>\n\u7f3a\u7701\u60c5\u51b5\u4e0b\u4f7f\u7528 `sys.stderr` \u8f93\u51fa\u5f02\u5e38\u4fe1\u606f\u3002\n\n__\u53c2\u6570 `ereturn`__<br>\n\u82e5\u88ab\u88c5\u9970\u7684\u51fd\u6570\u4e2d\u5f15\u53d1\u4e86\u5f02\u5e38\uff0c\u5c06\u8fd4\u56de\u6b64\u53c2\u6570\uff0c\u9ed8\u8ba4\u4e3a `None`\u3002<br>\n\u5b83\u5728\u67d0\u4e9b\u53ef\u4ee5\u8bbe\u5b9a\u975e `None` \u9ed8\u8ba4\u8fd4\u56de\u503c\u7684\u51fd\u6570\u4e2d\u975e\u5e38\u597d\u7528\u3002\n\n__\u53c2\u6570 `ecallback`__<br>\n\u63a5\u6536\u4e00\u4e2a\u53ef\u8c03\u7528\u5bf9\u8c61\uff0c\u82e5\u88ab\u88c5\u9970\u7684\u51fd\u6570\u4e2d\u5f15\u53d1\u4e86\u5f02\u5e38\u5c06\u8c03\u7528\u5b83\u3002<br>\n\u8fd9\u4e2a\u53ef\u8c03\u7528\u5bf9\u8c61\u8fd8\u9700\u63a5\u6536\u591a\u4e2a\u53c2\u6570\uff1a\u5f15\u53d1\u7684\u5f02\u5e38\u5bf9\u8c61\uff0c\u88ab\u88c5\u9970\u7684\u51fd\u6570\u5bf9\u8c61\uff0c\u88ab\u88c5\u9970\u7684\u51fd\u6570\u7684\u6240\u6709\u53c2\u6570\u3002\n\n__\u53c2\u6570 `eexit`__<br>\n\u8bbe\u4e3a `True` \u5c06\u5728\u5f15\u53d1\u5f02\u5e38\u540e\u629b\u51fa `SystemExit(4)`\uff0c\u5982\u679c\u6709 `ecallback` \u5219\u4f1a\u5148\u6267\u884c `ecallback`\u3002\n\n\n### \u4f7f\u7528\u88c5\u9970\u5668 `Retry` \u91cd\u8bd5\u51fd\u6570\u4e2d\u5f15\u53d1\u7684\u5f02\u5e38\n```python\nfrom gqylpy_exception import Retry\n\n@Retry(count=3, cycle=1)\ndef func():\n    int('a')\n```\n\u82e5\u88ab\u88c5\u9970\u7684\u51fd\u6570\u4e2d\u5f15\u53d1\u4e86\u5f02\u5e38\uff0c\u4f1a\u5c1d\u8bd5\u91cd\u65b0\u6267\u884c\u88ab\u88c5\u9970\u7684\u51fd\u6570\uff0c\u9ed8\u8ba4\u91cd\u8bd5 `Exception` \u53ca\u5176\u5b50\u7c7b\u7684\u6240\u6709\u5f02\u5e38\u3002\n\u50cf\u4e0a\u9762\u8fd9\u6837\u8c03\u7528 `Retry(count=3, cycle=1)` \u8868\u793a\u6700\u5927\u6267\u884c3\u6b21\uff0c\u6bcf\u6b21\u95f4\u96941\u79d2\u3002\u5b8c\u6574\u7684\u53c2\u6570\u5982\u4e0b\uff1a\n```python\ndef Retry(\n        etype:          Union[type, Tuple[type, ...]] = Exception,\n        *,\n        count:          int                           = inf,\n        cycle:          Union[int, float]             = 0,\n        ignore:         bool                          = False,\n        output_raw_exc: bool                          = False,\n        logger:         logging.Logger                = None\n):\n    ...\n```\n`Retry` \u7ee7\u627f `TryExcept`\uff0c\u4f60\u53ef\u4ee5\u5728 `TryExcept` \u4e2d\u627e\u5230\u53c2\u6570\u8bf4\u660e\uff0c\u4f46\u6ce8\u610f `Retry` \u8c03\u7528\u65e5\u5fd7\u8bb0\u5f55\u5668\u7684 `warning` \u65b9\u6cd5\u3002\n\n\u7ed3\u5408 `TryExcept` \u4f7f\u7528\uff0c\u65e2\u80fd\u91cd\u8bd5\u5f02\u5e38\u53c8\u80fd\u5904\u7406\u5f02\u5e38\uff1a\n```python\nfrom gqylpy_exception import TryExcept, Retry\n\n@TryExcept(ValueError)\n@Retry(count=3, cycle=1)\ndef func():\n    int('a')\n```\n\n\n",
            "description_content_type": "text/markdown",
            "docs_url": null,
            "download_url": "",
            "downloads": {
                "last_day": -1,
                "last_month": -1,
                "last_week": -1
            },
            "home_page": "http://gqylpy.com",
            "keywords": "",
            "license": "Apache 2.0",
            "maintainer": "",
            "maintainer_email": "",
            "name": "gqylpy-exception",
            "package_url": "https://pypi.org/project/gqylpy-exception/",
            "platform": null,
            "project_url": "https://pypi.org/project/gqylpy-exception/",
            "project_urls": {
                "Homepage": "http://gqylpy.com",
                "Source": "https://github.com/gqylpy/gqylpy-exception"
            },
            "release_url": "https://pypi.org/project/gqylpy-exception/1.3/",
            "requires_dist": null,
            "requires_python": ">=3.6, <4",
            "summary": "\u5728\u6267\u884c raise \u8bed\u53e5\u7684\u540c\u65f6\u521b\u5efa\u5f02\u5e38\u7c7b\uff0c\u65e0\u9700\u4e8b\u5148\u5b9a\u4e49\u5f02\u5e38\u7c7b\uff0c\u65b9\u4fbf\u5feb\u6377\u3002",
            "version": "1.3",
            "yanked": false,
            "yanked_reason": null
        },
        "last_serial": 15750466,
        "urls": [
            {
                "comment_text": "",
                "digests": {
                    "md5": "cf60c72c4926998ba20b12a3fba711d7",
                    "sha256": "b93711ab233020a94639a44ef29b51beba0339d88a261864dfd9eb23c130f598"
                },
                "downloads": -1,
                "filename": "gqylpy_exception-1.3-py3-none-any.whl",
                "has_sig": false,
                "md5_digest": "cf60c72c4926998ba20b12a3fba711d7",
                "packagetype": "bdist_wheel",
                "python_version": "py3",
                "requires_python": ">=3.6, <4",
                "size": 13105,
                "upload_time": "2022-11-13T10:24:52",
                "upload_time_iso_8601": "2022-11-13T10:24:52.294283Z",
                "url": "https://files.pythonhosted.org/packages/dc/cc/19c47f13b7c35927cce61a54520fc9fd39a87f46d6e5e1dbd2c896dce1e5/gqylpy_exception-1.3-py3-none-any.whl",
                "yanked": false,
                "yanked_reason": null
            }
        ],
        "vulnerabilities": []
    }
}